### PR TITLE
Proper API degradation for non-specialized cases

### PR DESCRIPTION
**Pull request by permcody**
This PR allows users to use distance and bounds routines that used to
only work when periodic and orthogonal meshes where used (detected).

refs #11939

### CODE REVIEW

File: framework/src/mesh/MooseMesh.C [hunk 1]
```diff
@@ -1372,11 +1372,10 @@ MooseMesh::detectOrthogonalDimRanges(Real tol)
 
   // See if we matched all of the extremes for the mesh dimension
   this->comm().max(extreme_matches);
-  if (std::count(extreme_matches.begin(), extreme_matches.end(), true) != std::pow(2.0, (int)dim))
-    return false; // This is not a regular orthogonal mesh
+  if (std::count(extreme_matches.begin(), extreme_matches.end(), true) == std::pow(2.0, (int)dim))
```
Comment (jwpeterson) [inline framework/src/mesh/MooseMesh.C]: Hmm... I think this line is doing a potentially unsafe floating point equality comparison. If you look at [the docs for std::pow](https://en.cppreference.com/w/cpp/numeric/math/pow), version (7) says "If any argument has integral type, it is cast to double" and the return type is either double or long double. I'd change this to something like `Utility::pow<dim>(2)` which is guaranteed to return an integral type.

**Review by jwpeterson — CHANGES_REQUESTED** [review]

File: framework/src/mesh/MooseMesh.C (hunk 1 above)
Comment (dschwen) [reply framework/src/mesh/MooseMesh.C]: Is dim known at compile time? If not just do 1 << dim

Comment (jwpeterson) [reply framework/src/mesh/MooseMesh.C]: Good point, `dim` is definitely a runtime variable.

**Comment by permcody** [comment]
Ping - comments addressed

**Review by jwpeterson — APPROVED** [review]

**Pull request merged**